main/21 Vinzari/Tab.62 Cumulativ.sql

```diff
@@ -0,0 +1,216 @@
+  FOR CR IN C
+  LOOP
+   IF CR.pCOD_CUATM = '0000000' THEN
+    INSERT INTO TABLE_OUT 
+(
+  PERIOADA,
+  FORM,
+  FORM_VERS,
+  ID_MDTABLE,
+  COD_CUATM,
+  NR_SECTIE,
+  NUME_SECTIE,
+  NR_SECTIE1,
+  NUME_SECTIE1,
+  NR_SECTIE2,
+  NUME_SECTIE2,
+  NR_ROW,
+  ORDINE,
+  DECIMAL_POS,
+  NUME_ROW,  
+ COL1,COL2,COL3,COL4,COL5
+)
+
+SELECT
+  :pPERIOADA AS PERIOADA,
+  :pFORM AS FORM,
+  :pFORM_VERS AS FORM_VERS,
+  :pID_MDTABLE AS ID_MDTABLE,
+  :pCOD_CUATM AS COD_CUATM,
+         
+  '0' AS NR_SECTIE,
+  '0' AS NUME_SECTIE,
+  '0' AS NR_SECTIE1,
+  '0' AS NUME_SECTIE1,
+  '0' AS NR_SECTIE2,
+  '0' AS NUME_SECTIE2,
+  NUME_ROW||'~'||NR_ROW,
+  ORDINE,
+  '02002' AS DECIMAL_POS,
+  NR_ROW,
+  COL1,
+  COL1/NOZERO(NUME_ROW) AS COL2,
+  COL3,
+  COL4,
+  COL4/NOZERO(COL3) AS COL5
+FROM
+(
+SELECT 
+  
+  R.NUME_ROW AS NR_ROW,
+  TO_NUMBER(R.NR_ROW) AS ORDINE,
+  SUM(CASE WHEN D.RIND IN ('65') AND CC.FULL_CODE LIKE '%'||R.CS_CUATM||'%' THEN D.COL1 END) AS NUME_ROW,
+  SUM(CASE WHEN D.RIND IN ('65') AND CC.FULL_CODE LIKE '%'||R.CS_CUATM||'%' THEN D.COL2 END) AS COL1,
+  SUM(CASE WHEN D.RIND IN ('65') AND CC.FULL_CODE LIKE '%'||R.CS_CUATM||'%' THEN D.COL4 END) AS COL3,
+  SUM(CASE WHEN D.RIND IN ('65') AND CC.FULL_CODE LIKE '%'||R.CS_CUATM||'%' THEN D.COL5 END) AS COL4
+FROM 
+  VW_DATA_ALL D
+--  INNER JOIN RENIM RN ON (D.CUIIO = RN.CUIIO AND D.CUIIO_VERS = RN.CUIIO_VERS)
+  INNER JOIN VW_CL_CUATM CC ON(D.CUATM = CC.CODUL)
+  CROSS JOIN 
+  (
+    SELECT
+      DECODE(CS.CODUL, 
+      '0000000','01','0100000','02','1111111','03','0300000','04',
+      '1400000','05','3400000','06','3600000','07','4100000','08','4300000','09',
+      '4500000','10','4800000','11','6200000','12','7100000','13','7400000','14',
+      '7800000','15','2222222','16','1000000','17','2500000','18','3100000','19',
+      '3800000','20','5300000','21','5500000','22','6000000','23','6400000','24',
+      '6700000','25','8000000','26','8300000','27','8900000','28','9200000','29',
+      '3333333','30','1200000','31','1700000','32','2100000','33','2700000','34',
+      '2900000','35','5700000','36','8500000','37','8700000','38','9600000','39')
+      AS NR_ROW,
+      CS.DENUMIRE AS NUME_ROW,
+      CS.CODUL AS CS_CUATM
+    FROM
+      VW_CL_CUATM CS
+    WHERE
+      CS.CODUL 
+      IN(
+         '0000000','0100000','1111111','0300000','1400000','3400000','3600000','4100000','4300000','4500000',
+         '4800000','6200000','7100000','7400000','7800000','2222222','1000000','2500000','3100000','3800000',
+         '5300000','5500000','6000000','6400000','6700000','8000000','8300000','8900000','9200000','3333333',
+         '1200000','1700000','2100000','2700000','2900000','5700000','8500000','8700000','9600000'
+        )
+  ) R 
+WHERE
+  D.FORM IN (26)                 AND 
+  D.FORM_VERS = :pFORM_VERS     AND    
+  D.CAPITOL IN (367)            AND 
+--  D.PERIOADA=:pPERIOADA AND
+ (D.PERIOADA BETWEEN FLOOR(:pPERIOADA/4)*4 AND :pPERIOADA) AND  
```

Comment:
--  D.PERIOADA=:pPERIOADA AND
 (D.PERIOADA BETWEEN FLOOR(:pPERIOADA/4)*4 AND :pPERIOADA) AND